frontend/src/app/modules/router/openproject.routes.ts

```diff
@@ -133,6 +135,15 @@ export function initializeUiRouterListeners(injector:Injector) {
         return $state.target(transition.to(), paramsCopy);
       }
 
+      // Set backRoute to know where we came from
+      if (fromState.name &&
+          fromState.data &&
+          toState.data &&
+          fromState.data.parent !== toState.data.parent) {
+          const paramsFromCopy = Object.assign({}, transition.params('from'));
+          backRoutingService.setBackRoute({ name: fromState.name, params: paramsFromCopy, parent: fromState.data.parent });
```

Comment:
Could you move this entire block into the backRoutingService and reduce the interface here to something like `backRoutingService.sync(fromState, toState)`? This avoids getting this initializer too convoluted over time.